pkg/controller/factory/factory.go

```diff
@@ -119,22 +124,54 @@ func (f *Factory) WithBareInformers(informers ...Informer) *Factory {
 // Pass the queueKeyFn you want to use to transform the informer runtime.Object into string key used by work queue.
 func (f *Factory) WithInformersQueueKeyFunc(queueKeyFn ObjectQueueKeyFunc, informers ...Informer) *Factory {
 	f.informerQueueKeys = append(f.informerQueueKeys, informersWithQueueKey{
-		informers:  informers,
-		queueKeyFn: queueKeyFn,
+		informers: informers,
+		queueKeysFn: func(obj runtime.Object) []string {
```

Comment:
Do we want to change the existing method, or add a new one?

Reply:
I think this one is not exposed. We will still maintain the existing method in factory.

Reply:
I would refactor this to:

```
func WithInformersSingleQueueKeyFunc()
func WithInformersMultipleQueueKeysFunc()
```

Reply:
change to WithInformersMultipleQueueKeysFunc(), should we keep the name of WithInformersQueueKeyFunc for compatible?